x-pack/legacy/plugins/maps/public/elasticsearch_geo_utils.js

```diff
@@ -203,118 +235,134 @@ export function geoShapeToGeometry(value, accumulator) {
+    })
+    : getEsSpatialRelationLabel(relation);
+  const meta = {
+    negate: false,
+    index: indexPatternId,
+    alias: `${geoFieldName} ${relationLabel} ${geometryLabel}`
+  };
+
+  if (geoFieldType === ES_GEO_FIELD_TYPE.GEO_SHAPE) {
     return {
+      meta,
       geo_shape: {
+        ignore_unmapped: true,
         [geoFieldName]: {
-          shape: safePolygon,
-          relation: 'INTERSECTS'
+          shape: geometry,
+          relation
         }
       }
     };
-  } else {
-    const errorMessage = i18n.translate('xpack.maps.elasticsearch_geo_utils.extent.unsupportedGeoFieldTypeErrorMessage', {
-      defaultMessage: `Unsupported field type, expected: geo_shape or geo_point, you provided: {geoFieldType}`,
-      values: { geoFieldType }
-    });
-    throw new Error(errorMessage);
   }
-}
 
+  // geo_points supports limited geometry types
+  // TODO add support for multi_polygon
+  ensureGeometryType(geometry.type, [GEO_JSON_TYPE.POLYGON]);
 
-export function createExtentFilterWithMeta(mapExtent, indexPatternId, geoFieldName, geoFieldType) {
-
-  const roundedExtent = {
-    minLon: _.round(mapExtent.minLon, DECIMAL_DEGREES_PRECISION),
-    minLat: _.round(mapExtent.minLat, DECIMAL_DEGREES_PRECISION),
-    maxLon: _.round(mapExtent.maxLon, DECIMAL_DEGREES_PRECISION),
-    maxLat: _.round(mapExtent.maxLat, DECIMAL_DEGREES_PRECISION)
-  };
+  if (isBoundingBox) {
+    return createGeoBoundBoxFilter(geometry, geoFieldName, { meta });
+  }
 
-  const filter = createExtentFilter(roundedExtent, geoFieldName, geoFieldType);
-  filter.meta = {
-    negate: false,
-    index: indexPatternId,
-    alias: i18n.translate('xpack.maps.elasticsearch_geo_utils.extentFilter.aliasTitle', {
-      defaultMessage: `extent at {coordinate}`,
-      values: {
-        coordinate: `[${roundedExtent.minLon}, ${roundedExtent.minLat}, ${roundedExtent.maxLon}, ${roundedExtent.maxLat}]`
+  return {
+    meta,
+    geo_polygon: {
+      ignore_unmapped: true,
+      [geoFieldName]: {
+        points: geometry.coordinates[POLYGON_COORDINATES_EXTERIOR_INDEX].map(coordinatePair => {
+          return {
+            lon: coordinatePair[LON_INDEX],
+            lat: coordinatePair[LAT_INDEX]
+          };
+        })
       }
-    })
+    }
   };
-  return filter;
 }
 
-export function createShapeFilterWithMeta(geojsonPolygon, indexPatternId, geoFieldName, geoFieldType) {
-
-  const filter = {
-    meta: {
-      negate: false,
-      index: indexPatternId,
-      alias: i18n.translate('xpack.maps.elasticsearch_geo_utils.shapeFilter.aliasTitle', {
-        defaultMessage: `shape at {coordinate}`,
-        values: {
-          // eslint-disable-next-line max-len
-          coordinate: `${_.round(geojsonPolygon.coordinates[0][0][0], DECIMAL_DEGREES_PRECISION)}, ${_.round(geojsonPolygon.coordinates[0][0][1], DECIMAL_DEGREES_PRECISION)}`
-        }
-      })
+export function roundCoordinates(coordinates, precision = DECIMAL_DEGREES_PRECISION) {
```

Comment:
remove overloaded second parameter. It is never called with anything other then DECIMAL_DEGREES_PRECISION.